pkg/tests/cross-tests/cross_test.go

```diff
@@ -403,3 +405,287 @@ func TestAws2442(t *testing.T) {
+				"regex_pattern_set_reference_statement": []interface{}{},
+				"rule_group_reference_statement":        []interface{}{},
+				"size_constraint_statement":             []interface{}{},
+				"sqli_match_statement":                  []interface{}{},
+				"xss_match_statement":                   []interface{}{}}},
+			"visibility_config": []interface{}{map[string]interface{}{
+				"cloudwatch_metrics_enabled": true,
+				"metric_name":                "IPAllowRule",
+				"sampled_requests_enabled":   true,
+			}}}
+
+		expected2 := map[string]interface{}{
+			"action": []interface{}{map[string]interface{}{
+				"allow": []interface{}{},
+				"block": []interface{}{
+					map[string]any{
+						"custom_response": []interface{}{},
+					},
+				},
+				"captcha":   []interface{}{},
+				"challenge": []interface{}{},
+				"count":     []interface{}{}}},
+			"captcha_config":  []interface{}{},
+			"name":            "IPAllowRule",
+			"override_action": []interface{}{},
+			"priority":        0,
+			//"rule_label":      schema.NewSet(nil, nil),
+			"statement": []interface{}{map[string]interface{}{
+				"and_statement":        []interface{}{},
+				"byte_match_statement": []interface{}{},
+				"geo_match_statement":  []interface{}{},
+				"ip_set_reference_statement": []interface{}{map[string]interface{}{
+					"arn":                        "some-arn",
+					"ip_set_forwarded_ip_config": []interface{}{},
+				}},
+				"label_match_statement":                 []interface{}{},
+				"managed_rule_group_statement":          []interface{}{},
+				"not_statement":                         []interface{}{},
+				"or_statement":                          []interface{}{},
+				"rate_based_statement":                  []interface{}{},
+				"regex_match_statement":                 []interface{}{},
+				"regex_pattern_set_reference_statement": []interface{}{},
+				"rule_group_reference_statement":        []interface{}{},
+				"size_constraint_statement":             []interface{}{},
+				"sqli_match_statement":                  []interface{}{},
+				"xss_match_statement":                   []interface{}{}}},
+			"visibility_config": []interface{}{map[string]interface{}{
+				"cloudwatch_metrics_enabled": true,
+				"metric_name":                "IPAllowRule",
+				"sampled_requests_enabled":   true,
+			}}}
+
+		switch {
+		case assert.ObjectsAreEqual(expected, i):
+			fmt.Printf("\n\n#### Computing hash set for rule <<expected>> (action=%#v, ruleLabel=%#v)==> %d\n\n", action, ruleLabel, actual)
+		case assert.ObjectsAreEqual(expected2, i):
+			fmt.Printf("\n\n#### Computing hash set for rule <<expected2>> (action=%#v, ruleLabel=%#v)==> %d\n\n", action, ruleLabel, actual)
+		default:
+			assert.Equal(t, expected, i)
+		}
+		return actual
+	}
+
+	runDiffCheck(t, diffTestCase{
+		Resource:   resource,
+		Config1:    cfg,
+		Config2:    cfg,
+		SkipPulumi: false,
+	})
+}
+
+func TestAws3880Minimal(t *testing.T) {
```

Comment:
That's quite a few levels of nesting - if required then we need to update the property based tests to go deeper - this could not have been generated there.